x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/IdentifierBuilder.java

```diff
@@ -63,17 +64,36 @@ public String visitIndexPattern(List<EsqlBaseParser.IndexPatternContext> ctx) {
         ctx.forEach(c -> {
             String indexPattern = visitIndexString(c.indexString());
             String clusterString = c.clusterString() != null ? c.clusterString().getText() : null;
+            String selectorString = c.selectorString() != null ? c.selectorString().getText() : null;
             // skip validating index on remote cluster, because the behavior of remote cluster is not consistent with local cluster
             // For example, invalid#index is an invalid index name, however FROM *:invalid#index does not return an error
             if (clusterString == null) {
                 hasSeenStar.set(indexPattern.contains(WILDCARD) || hasSeenStar.get());
                 validateIndexPattern(indexPattern, c, hasSeenStar.get());
             }
-            patterns.add(clusterString != null ? clusterString + REMOTE_CLUSTER_INDEX_SEPARATOR + indexPattern : indexPattern);
+            if (selectorString != null) {
+                IndexNameExpressionResolver.SelectorResolver.validateIndexSelectorString(indexPattern, selectorString);
```

Comment:
Here you should catch whatever exception the validation mechanism is throwing and wrap it in a ParsingException. See, for example, https://github.com/elastic/elasticsearch/pull/120660/files#diff-2ffef2b3bfd7e9a06759cd92489858b35223bb23966842838961ee19eb62bf91R117-R119

There are some test examples for this exception check in `StatementParserTests.testInvalidCharacterInIndexPattern`